exist-core/src/main/java/org/exist/dom/persistent/Match.java

```diff
@@ -385,23 +390,17 @@ public static boolean matchListEquals(final Match m1, final Match m2) {
     }
 
     @Override
-    public boolean equals(final Object other) {
-        if(!(other instanceof Match)) {
-            return false;
+    public boolean equals(final Object obj) {
+        if (obj == this) {
+            return true;
         }
-        final Match om = (Match) other;
-        return om.matchTerm != null &&
-            om.matchTerm.equals(matchTerm) &&
-            om.nodeId.equals(nodeId);
-    }
 
-    public boolean matchEquals(final Match other) {
-        if(this == other) {
-            return true;
+        if (obj instanceof Match other) {
```

Comment:
not sure if  Iunderstand this line @reinhapa

Reply:
@dizzzz new feature starting with JDK 17 auto cast to variable "other" in case non null and matching instance...